src/PerpetualPositionRouter.sol

```diff
@@ -139,12 +149,23 @@ contract PerpetualPositionRouter {
   //
   // 1. What us a reasonable amount of precision to reduce the function?
   // 2. What are reasonable time periods for deadlines
-  //
-  // closePosition is not using the amount value and maybe we could optimize by splitting it into a
-  // separate contract
   fallback() external payable {
-    (uint8 funcName, uint256 amount, uint256 oppositeAmountBound, uint160 sqrtPriceLimitX96) =
-      abi.decode(msg.data, (uint8, uint256, uint256, uint160));
+    // The first 11 bytes are padding.
+    // The 12th byte will contain the function name
+    // which we will use to pick the correct decoding
+    // strategy
+    uint8 funcName = uint8(bytes1(msg.data[11:12]));
```

Comment:
Why not just pull this out of combinedArgs as well?

Reply:
I may be misunderstanding are you saying pull combinedArgs from the msg.data and then pull out the funcName?

The main reason I pulled this out first was to conditionally pick the decoding strategy. Is there a better way you have in mind of doing this?